tests/unittests/MLTest.cpp

```diff
@@ -205,6 +204,82 @@ TEST_P(MLTest, learnXor) {
   }
 }
 
+/// Learn the logarithmic function.
+TEST_P(MLTest, learnLog) {
+  unsigned numInputs = 50;
+  unsigned batchSize = 7;
+  EE_.getConfig().learningRate = 0.07;
+  EE_.getConfig().batchSize = batchSize;
+
+  auto &mod = EE_.getModule();
+  auto &PRNG = mod.getPRNG();
+  Function *F = mod.createFunction("main");
+  F->setName("learnLog");
+
+  auto *A =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 1}, "A", VisibilityKind::Public,
+                         Variable::TrainKind::None);
+  auto *Ex =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 1}, "Ex", VisibilityKind::Public,
+                         Variable::TrainKind::None);
+
+  Node *O = F->createFullyConnected("fc1", A, 5);
+  O = F->createTanh("tanh1", O);
+  O = F->createFullyConnected("fc2", O, 5);
+  O = F->createTanh("tanh2", O);
+  O = F->createFullyConnected("fc3", O, 1);
+  O = F->createTanh("tanh3", O);
+  O = F->createRegression("reg", O, Ex);
+  auto *result = F->createSave("ret", O);
+
+  // Set the training data. 
+  Tensor trainingSet(ElemKind::FloatTy, {numInputs, 1});
+  Tensor trainingLabels(ElemKind::FloatTy, {numInputs, 1});
+
+  auto TS = trainingSet.getHandle<>();
+  auto TL = trainingLabels.getHandle<>();
+
+  // Set the training date as floating number from range [0.75, 1.5).
+  const float LO = 0.75; // Lower bound of training data 
+  const float HI = 1.5;  // Upper bound of training data
+  for (size_t i = 0; i < numInputs; i ++) {
+    // Generate a floating number in the range of [LO,HI)
+    float a = (LO+HI)/2 + PRNG.nextRand()*(HI-LO)/2;
+    TS.at({i,0}) = a;
+    TL.at({i,0}) = std::log(a);
+  }
+
+  Function *TF = glow::differentiate(F, EE_.getConfig());
+  EE_.compile(CompilationMode::Train, TF);
+
+  // Train the network:
+  EE_.runBatch(1000, {A, Ex}, {&trainingSet, &trainingLabels});
+
+  EE_.compile(CompilationMode::Infer, F);
+
+  // Set the testing data.
+
+  Tensor testSet(ElemKind::FloatTy, {batchSize, 1});
+
+  auto TES = testSet.getHandle<>();
+
+  for (size_t i = 0; i < batchSize; i++) {
+    // Generate a floating number in the range of [LO,HI)
+    float a = (LO+HI)/2 + PRNG.nextRand()*(HI-LO)/2;
+    TES.at({i,0}) = a;
+  }
+
+  EE_.run({A}, {&testSet});
+  auto resH = result->getVariable()->getPayload().getHandle<>();
+
+  // Test the output:
+  for (size_t i = 0; i < batchSize; i++) {
+    float a = TES.at({i, 0});
+    EXPECT_NEAR(resH.at({i, 0}), (std::log(a)), 0.02);
```

Comment:
Also curious, what kind of results do you see here? I.e. how close of a log approximate does it produce?

Reply:
The errors between predicted and test label are kinda below 0.015 now.  I tried to restrain it to be below 0.01. But still there are some cases can not reach this narrow error.